Source/WebCore/dom/Element.cpp

```diff
@@ -5107,8 +5107,43 @@ void Element::requestFullscreen(FullscreenOptions&& options, RefPtr<DeferredProm
 #else
     bool optionsEnabled = document().settings().fullScreenKeyboardLock();
 #endif
-    if (optionsEnabled && document().page() && document().page()->hardwareKeyboardAttached())
+
+    RefPtr page = document().page();
+    if (optionsEnabled && page && page->chrome().client().isHardwareKeyboardAttached()) {
+        auto& client = document().page()->chrome().client();
+
+        // Registers a callback to exit fullscreen mode
+        client.startHardwareKeyboardAttachmentNotifications([weakThis = WeakPtr { *this }](bool attached) {
+            auto protectedThis = weakThis.get();
+            if (!protectedThis)
+                return;
+
+            // Exit the fullscreen API when the hardware keyboard is detached.
+            if (!attached) {
+                protectedThis->protectedDocument()->postTask(
+                    WTF::Function<void()>([weakThis]() {
```

Comment:
Do you need to explicitly create `WTF::Function`? I feel like any lambda you pass in should work? You could try this:

```
protectedThis->protectedDocument()->postTask([weakThis] {
    ...
});
```

Reply:
if you have to explicitly specify `WTF::Function`, can you remove the `WTF` prefix? Typically we do not use that namespace prefix in the project.

Reply:
Also, last nit, you can skip empty lambda arg lists in C++23
```suggestion
                    WTF::Function<void()>([weakThis] {
```